packages/manager/apps/dedicated/client/app/ip/ip/ip-ip.controller.js

```diff
@@ -113,12 +123,51 @@ export default /* @ngInject */
     }
   });
 
-  $scope.displayOrganisation = function displayOrganisation() {
-    trackClick('manage-organisation');
-    goToOrganisation().then(() =>
-      $scope.$broadcast('ips.organisation.display'),
-    );
-  };
+  const debouncedReload = debounce(() => {
+    $timeout(() => $scope.$broadcast('ips.table.reload'));
+  }, 200);
+
+  // State management & grid reload
+  // React to both user selections and url changes (including back / force mecanism)
+  $scope.$watch(
+    () => {
+      const { ipService, serviceName, serviceType } = $location.search();
+      return {
+        selection: $scope.selection,
+        params: { ipService, serviceName, serviceType },
+      };
+    },
+    ({ params: { ipService, serviceName, serviceType } }) => {
+      if (ipService && !ipServiceData) {
+        $state.go(
+          'app.ip',
+          { ipService, serviceType: null, serviceName: null },
+          { reload: true },
+        );
+        return;
```

Comment:
```suggestion
        return $state.go(
          'app.ip',
          { ipService, serviceType: null, serviceName: null },
          { reload: true },
        );
```

Reply:
Don't need to return anything inside a $watch handler